src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Agent/AgentManager.cs

```diff
@@ -0,0 +1,117 @@
+using Azure.AI.Agents.Persistent;
+using Azure.Tools.GeneratorAgent.Configuration;
+using Microsoft.Extensions.Logging;
+
+namespace Azure.Tools.GeneratorAgent.Agent
+{
+    internal class AgentManager : IAsyncDisposable
```

Comment:
tbh I feel like this could probably be kept in AgentOrchestrator, it looks like it's essentially a wrapper around PersistentAgentsAdministrationClient, but it seems light enough that instead just calling create/delete/update directly on the admin client shouldn't be much more overhead. And the orchestrator could instead manage the PersistantAgent? Idk let me know what you think about that idea